sale_order_type/models/sale_order.py

```diff
@@ -31,6 +31,12 @@ def onchange_type_id(self):
         self.warehouse_id = self.type_id.warehouse_id
         self.picking_policy = self.type_id.picking_policy
         self.order_policy = self.type_id.order_policy
+        if self.type_id.payment_term_id:
+            self.payment_term = self.type_id.payment_term_id.id
+        if self.type_id.pricelist_id:
+            self.pricelist_id = self.type_id.pricelist_id.id
+        if self.type_id.incoterm_id:
+            self.incoterm = self.type_id.incoterm_id.id
```

Comment:
@tafaRU , I'm sorry I made a mistake:

Being this method decorated with the onchange decorator, it will not trigger the write method of the object, instead all the work will be done in the cache. No performance issue in this case.
I'm sorry for wasting your time.